### PR TITLE
docs(hydroflow_plus): add initial docs on consistency and safety

### PR DESCRIPTION
**Pull request by shadaj**

### CODE REVIEW

File: hydroflow_plus_test/src/cluster/mod.rs [hunk 1]
```diff
@@ -8,3 +8,9 @@ pub mod quorum;
 pub mod request_response;
 pub mod simple_cluster;
 pub mod two_pc;
+
+#[stageleft::runtime]
+pub mod docs {
+    #[doc = include_str!("../../../docs/docs/hydroflow_plus/consistency.mdx")]
```
Comment (MingweiSamuel) [inline hydroflow_plus_test/src/cluster/mod.rs:14]: the mdx syntax won't look quite right

File: docs/docs/hydroflow_plus/consistency.mdx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+A key feature of Hydroflow+ is its integration with the Rust type system to highlight possible sources of inconsistent distributed behavior due to sources of non-determinism such as batching, timeouts, and message reordering. In this section, we'll walk through the consistency guarantees in Hydroflow+ and how to use the **`unsafe`** keyword as an escape hatch when introducing sources of non-determinism.
+
+:::note
+
+Our consistency and safety model is based on the POPL'25 paper [Flo: A Semantic Foundation for Progressive Stream Processing](https://arxiv.org/abs/2411.08274), which covers the formal details and proofs underlying this system.
+
+:::
+
+## Determinism and Eventual Consistency
+Much existing literature in distributed systems focuses on consistency levels such as "eventual consistency" which typically correspond to guarantees when reading the state of an object (or set of objects) at a _specific point_ in time. Hydroflow+ does not use such a consistency model internally, since it uses a streaming model where values are being continuously updated _over time_.
```
Comment (MingweiSamuel) [inline docs/docs/hydroflow_plus/consistency.mdx:15]: Seems like apples to oranges, eventual consistency really is just for replicated data? Determinism seems like a different thing

Comment (MingweiSamuel) [reply docs/docs/hydroflow_plus/consistency.mdx:15]: Maybe would be better to have this paragraph after/tied to "eventual determinism," seems analogous

**Review by MingweiSamuel — COMMENTED** [review]
Maybe add a note about the determinism of rust closure UDFs too? `map(q!(|_| random()))` would slip through right now?

**Review by MingweiSamuel — APPROVED** [review]

File: hydroflow_plus_test/src/cluster/mod.rs (hunk 1 above)
Comment (shadaj) [reply hydroflow_plus_test/src/cluster/mod.rs:14]: Ah, this is meant to be hidden! Just to get it to be processed in doctests. Will add a `#[doc(hidden)]`

**Pull request merged**